src/main/java/randoop/reflection/TypeInstantiator.java

```diff
@@ -98,6 +99,140 @@ public TypedClassOperation instantiate(TypedClassOperation operation) {
+   */
+  public Set<TypedClassOperation> instantiateWithMultipleTypes(TypedClassOperation operation) {
+    assert operation.isGeneric() || operation.hasWildcardTypes()
+        : "operation must be generic or have wildcards";
+
+    // This list is the same size as the set returned by this method.
+    List<Substitution> substitutions = new ArrayList<>();
+
+    ClassOrInterfaceType declaringType = operation.getDeclaringType();
+    // If the declaring type of the operation is generic, get the set of possible instantiations.
+    if (declaringType.isGeneric()) {
+
+      // If the operation creates objects of its declaring type, it may create a new instantiation.
+      if (operation.isConstructorCall()
+          || (operation.isStatic() && operation.getOutputType().equals(declaringType))) {
+        if (declaringType.isSubtypeOf(JDKTypes.SORTED_SET_TYPE)) {
+          instantiationsForSortedSetType(operation, substitutions);
+        } else {
+          instantiationsForGeneralTypes(declaringType, substitutions);
+        }
+      } else {
+        // Otherwise, select from existing ones.
+        for (Type type : inputTypes) {
+          if (type.isParameterized()
+              && ((InstantiatedType) type).isInstantiationOf(declaringType)) {
+            substitutions.add(
+                ((InstantiatedType) type).getInstantiatingSubstitution(declaringType));
+          }
+        }
+      }
+    }
+
+    Set<TypedClassOperation> result = new HashSet<>();
+    // Apply each one of these type substitutions and create an instantiated operation.
+    for (Substitution substitution : substitutions) {
+      if (substitution != null) {
+        // Instantiate the type parameters of the declaring type.
+        TypedClassOperation typedOperation = operation.substitute(substitution);
+
+        // If the operation includes wildcard types, do capture conversion first.
+        if (typedOperation.hasWildcardTypes()) {
+          Log.logPrintf("Applying capture conversion to %s%n", typedOperation);
+          typedOperation = typedOperation.applyCaptureConversion();
+        }
+
+        typedOperation = instantiateOperationTypes(typedOperation);
+        if (typedOperation != null) {
+          result.add(typedOperation);
+        }
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Find instantiations for types that are not subtypes of {@code SortedSet}.
+   *
+   * @param declaringType the type to find substitutions for
+   * @param substitutions set of substitutions. This method modifies this set by adding to it.
+   */
+  private void instantiationsForGeneralTypes(
+      ClassOrInterfaceType declaringType, List<Substitution> substitutions) {
+    List<TypeVariable> typeParameters = declaringType.getTypeParameters();
+    List<Substitution> possSubs = allExtendingSubstitutions(typeParameters, new Substitution());
+
+    // For each possible substitution, check that applying such a substitution will give us a fully
```

Comment:
Why wouldn't it?  What should be done if it is not?

Reply:
Don't understand code well enough to answer.